workflow/src/main/java/com/synopsys/integration/alert/workflow/message/NotificationReceiver.java

```diff
@@ -82,10 +88,14 @@ public void handleEvent(NotificationReceivedEvent event) {
             logger.trace("Processing Page: {}. New pages found: {}",
                 numPagesProcessed,
                 pageOfAlertNotificationModels.getTotalPages());
+            hasNotTimedOut = DateUtils.createCurrentDateTimestamp().isBefore(timeLimit);
         }
         if (numPagesProcessed == MAX_NUMBER_PAGES_PROCESSED) {
             logger.warn("Receiver reached upper page limit of pages processed: {}, exiting.", MAX_NUMBER_PAGES_PROCESSED);
         }
+        if (!hasNotTimedOut) {
```

Comment:
This double negative is a bit confusing. Can we rename the variable in the positive so the logic surrounding it is easier to follow?

Reply:
If we make the change you are suggesting then the while condition will have to be negated. So I think this is easier to follow.

Reply:
Or are you saying, that would be easier to follow because the "Not" will be removed from the variable name?

Reply:
I think I agree with Gavin. If we change it to hasTimedOut=false, we would negate in the while loop to read as "has not timed out". And it would be easier to read in the if statement as "has timed out"